corporate/lib/stripe.py

```diff
@@ -969,6 +1009,81 @@ def make_end_of_cycle_updates_if_needed(
             return None, None
         return None, last_ledger_entry
 
+    def get_billing_page_context(self) -> Optional[Dict[str, Any]]:
+        customer = self.get_customer()
+        assert customer is not None
+        plan = get_current_plan_by_customer(customer)
+        if plan is not None:
+            now = timezone_now()
+            new_plan, last_ledger_entry = self.make_end_of_cycle_updates_if_needed(plan, now)
+            if last_ledger_entry is not None:
+                if new_plan is not None:  # nocoverage
+                    plan = new_plan
+                assert plan is not None  # for mypy
+                downgrade_at_end_of_cycle = plan.status == CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE
+                switch_to_annual_at_end_of_cycle = (
+                    plan.status == CustomerPlan.SWITCH_TO_ANNUAL_AT_END_OF_CYCLE
+                )
+                licenses = last_ledger_entry.licenses
+                licenses_at_next_renewal = last_ledger_entry.licenses_at_next_renewal
+                seat_count = self.current_count_for_billed_licenses()
+
+                # Should do this in JavaScript, using the user's time zone
+                if plan.is_free_trial():
+                    assert plan.next_invoice_date is not None
+                    renewal_date = "{dt:%B} {dt.day}, {dt.year}".format(dt=plan.next_invoice_date)
+                else:
+                    renewal_date = "{dt:%B} {dt.day}, {dt.year}".format(
+                        dt=start_of_next_billing_cycle(plan, now)
+                    )
+
+                renewal_cents = renewal_amount(plan, now, last_ledger_entry)
+                charge_automatically = plan.charge_automatically
+                assert customer.stripe_customer_id is not None  # for mypy
+                stripe_customer = stripe_get_customer(customer.stripe_customer_id)
+                if charge_automatically:
+                    payment_method = payment_method_string(stripe_customer)
+                else:
+                    payment_method = "Billed by invoice"
```

Comment:
Should we include `stripe_get_customer` as a static method?

Reply:
I'm not sure I understand the question here.

Reply:
The code block here
```py
    renewal_cents = renewal_amount(plan, now, last_ledger_entry)
    charge_automatically = plan.charge_automatically
    assert customer.stripe_customer_id is not None  # for mypy
    stripe_customer = stripe_get_customer(customer.stripe_customer_id)
    if charge_automatically:
        payment_method = payment_method_string(stripe_customer)
    else:
        payment_method = "Billed by invoice"
```
calls `stripe_get_customer`, that is
```py
@catch_stripe_errors
def stripe_get_customer(stripe_customer_id: str) -> stripe.Customer:
    return stripe.Customer.retrieve(
        stripe_customer_id, expand=["invoice_settings", "invoice_settings.default_payment_method"]
    )
```

Should we include such functions that make API call to stripe as a static method to BillingSession (as it is a helper method logically related to the class)?

Reply:
or keeping it as a global function is fine.

Reply:
I don't think it needs to be moved to `BillingSession`, since it wouldn't access any attributes of the class. I could imagine if we find that only ~20% of `corporate/lib/stripe.py` is outside that class, deciding to move other functions just because that feels more readable.